Kernel/Arch/aarch64/ASM_wrapper.h

```diff
@@ -51,4 +54,34 @@ inline ExceptionLevel get_current_exception_level()
     return static_cast<ExceptionLevel>(current_exception_level);
 }
 
+inline void wait_cycles(int n)
+{
+    // This is probably too fast when caching and branch prediction is turned on.
+    // FIXME: Make timer-based.
+    asm("0:\n"
+        "    subs %[value], %[value], #1\n"
+        "    bne 0b" ::[value] "r"(n));
+}
+
+inline void el1_vector_table_install(void* vector_table)
+{
+    asm("msr VBAR_EL1, %[value]" ::[value] "r"(vector_table));
+}
+
+inline void enter_el2_from_el3()
+{
+    asm("    adr x0, entered_el2\n"
```

Comment:
This clobbers x0. https://gcc.gnu.org/onlinedocs/gcc/Basic-Asm.html#Basic-Asm suggests that this might lead to subtle problems down the line. Maybe we should use the extended asm syntax for everything in this file (and probably in serenity in general)?

Reply:
Yep, I was indeed missing a clobber here. I also agree that we should be using the extended asm syntax.